protobuf/v1/snapshot.proto

```diff
@@ -0,0 +1,86 @@
+syntax = "proto3";
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/wrappers.proto";
+import "google/protobuf/timestamp.proto";
+import "nexus.proto";
+package mayastor.v1;
+
+service SnapshotRpc {
+    // Common snapshot related methods.
+    rpc CreateReplicaSnapshot(CreateReplicaSnapshotRequest) returns (CreateReplicaSnapshotResponse) {}
+    rpc CreateNexusSnapshot(NexusCreateSnapshotRequest) returns (NexusCreateSnapshotResponse) {}
+    rpc ListSnapshot(ListSnapshotsRequest) returns (ListSnapshotsResponse) {}
+    rpc DeleteSnapshot(DeleteSnapshotRequest) returns (google.protobuf.Empty) {}
+}
+
+/// Parameter details of a single snapshot
+message SnapshotInfo {
+    string snapshot_uuid = 1; // uuid of the snapshot
+    string snapshot_name = 2; // name of the snapshot
+    uint64 snapshot_size = 3; // amount of bytes referenced by snapshot
+    uint64 num_clones = 4; // number of clones created from this snapshot
+    google.protobuf.Timestamp timestamp = 5; // snapshot creation_timestamp
+    string source_uuid = 6; // uuid of the source from which snapshot is created (nexus / replica)
+    uint64 source_size = 7; // amount of bytes referenced by source
+    string entity_id = 8; // uuid of the entity for which snapshot is taken.
+    string txn_id = 9; // unique transaction id for snapshot.
+    bool valid_snapshot = 10; // true: valid, false: invalid (missing one/more snapshotdescriptor field i.e. txn_id, entity_id, source_uuid
+}
+/// Request parameters to list snapshot.
+message ListSnapshotsRequest {
```

Comment:
Isn't this Replica Specific?

Reply:
> Isn't this Replica Specific?

This will work for both replica and nexus. If `source_uuid` is given then all the snapshots for which the parent is `source_uuid` will be returned back. CP should ensure to send the request to specific io-engine node from which the information to be fetched.
If the `snapshot_uuid` is passed, then that specific snapshot information will be returned.